lib/chef/resource/locale.rb

```diff
@@ -67,10 +67,18 @@ def lc_all(arg = nil)
 
       action :update do
         description "Update the system's locale."
-        unless up_to_date?
-          converge_by "Updating System Locale" do
-            generate_locales unless unavailable_locales.empty?
-            update_locale
+        if windows?
+          unless new_resource.lang == get_system_locale
+            converge_if_changed :lang do
```

Comment:
`converge_if_changed` here isn't going to work if there's no load_current_value, this should be `converge_by` (and the conditional above `converge_if_changed` doesn't make any sense, either the pattern should be to use a conditional with `converge_by` or to use `converge_if_changed` without a conditional and with load_current_value).

Reply:
@lamont-granquist `new_resource.lang == get_system_locale` this condition checks if the user supplied the same `lang` value which is already set in the current system, so on that time, it should not be run `set_system_locale` method.